File: frontend/app/models/bet.js
import DS from 'ember-data';

export default DS.Model.extend({
  totalValue: DS.attr('number'),

  userId: DS.attr('number'),
  matchId: DS.attr('number'),
  teamId: DS.attr('number'),

  items: DS.hasMany('item'),

  user: DS.belongsTo('user'),
  match: DS.belongsTo('match'),
  team: DS.belongsTo('team')
});

File: app/models/bet.rb
class Bet < ActiveRecord::Base
  attr_accessor :payout
  validates :total_value, :team, presence: true
  validates :total_value, numericality: { greater_than: 9 }
  validates :items, length: { minumum: 1, maximum: 10 }

  validates_uniqueness_of :user, scope: :match

  belongs_to :user
  belongs_to :match
  belongs_to :team
  has_many :items

  @payout = 0
end
